Lib/linecache.py

```diff
@@ -5,10 +5,8 @@
 that name.
 """
 
-import functools
```

Comment:
These imports are moved to the functions to avoid problems that arise trying to import these modules too early (they help breaking some circular dependency)